### DISCUSSION TITLE
Per thread roles

### DISCUSSION THREAD

**Discussion by nvahalik** [RTippin/messenger#43] [Ideas]
Currently, each `Participant` within a `Thread` has individual permissions that can be assigned to them. Unless they have `admin` assigned, they permissions are always set here on the participant.

It would be helpful in some circumstances to have roles associated with the `Participant` model:

* Have permissions tied to roles.
* Be able to model permissions that depend on roles 

For example, I have a use case where I have teams of people that relate to people one-on-one. 90% of the time the `Participant`s of a thread will be the `user` and `team member`. But _sometimes_ we may have a `team admin` come into the conversation. In our case, we could look at some positional permission on the Participant to know which user is the `user` and which is the `team member` but it would be easier to know that the "regular user" in a `Thread` is actually the `Participant` with the `user` role.

**Comment by RTippin** [comment]
This sounds like an excellent idea! I will be working on version 2 soon, and will look into the best way to implement this.
